texar/modules/connectors/connectors.py

```diff
@@ -0,0 +1,866 @@
+
+        Here:
+
+        `"activation_fn"`: str
+            The activation function applied to the outputs of the MLP
+            transformation layer. Can be a function,
+            or its name or module path.
+        `"name"`: str
+            Name of the connector.
+        """
+        return {
+            "activation_fn": "texar.core.layers.identity",
+            "name": "reparameterized_stochastic_connector"
+        }
+
+    def forward(self,    # type: ignore
+                num_samples: Optional[Union[int, torch.Tensor]] = None,
+                transform: bool = True) -> Tuple[Any, Any]:
+        r"""Samples from a distribution and optionally performs transformation
+        with an MLP layer.
+        The distribution must be reparameterizable, i.e.,
+        :python:`Distribution.has_rsample == True`.
+
+        Args:
+            num_samples (optional): An ``int`` or ``int`` ``Tensor``.
+                Number of samples to generate. If not given,
+                generate a single sample. Note that if batch size has
+                already been included in :attr:`distribution`'s dimensionality,
+                :attr:`num_samples` should be left as ``None``.
+            transform (bool): Whether to perform MLP transformation of the
+                distribution samples. If ``False``, the structure/shape of a
+                sample must match :attr:`output_size`.
+
+
+        :returns:
+            A tuple (:attr:`output`, :attr:`sample`), where
+
+            - output: A ``Tensor`` or a (nested) ``tuple`` of Tensors with
+              the same structure and size of :attr:`output_size`.
+              The batch dimension equals :attr:`num_samples` if specified,
+              or is determined by the distribution dimensionality.
+              If :attr:`transform` is `False`, it will be
+              equal to :attr:`sample`.
+            - sample: The sample from the distribution, prior to transformation.
+
+            Otherwise, returns a ``Tensor`` :attr:`sample`, where
+            - sample: The sample from the distribution, prior to transformation.
+
+        Raises:
+            ValueError: If distribution cannot be reparameterizable.
+            ValueError: The output does not match :attr:`output_size`.
+        """
+        if isinstance(self._dstr_type, str):
+            dstr: Distribution = utils.check_or_get_instance(
+                self._dstr_type, self._dstr_kwargs,
+                ["torch.distributions", "texar.custom"])
+        else:
+            dstr = self._dstr_type
+
+        if not dstr.has_rsample:
+            raise ValueError("Distribution should be reparameterizable")
+
+        if num_samples:
+            sample = dstr.rsample([num_samples])
+        else:
+            sample = dstr.rsample()
+
+        if transform:
+            fn_modules = ['torch', 'torch.nn', 'texar.custom']
+            activation_fn = utils.get_function(
+                self.hparams.activation_fn, fn_modules)
```

Comment:
It's better to move this part into `__init__`. Also, we should probably include `torch.nn.functional` instead of `torch.nn` because we're using the functional APIs.

Reply:
And yeah, we have `texar.core.get_activation_fn`, why not just use that? Could you also change `get_activation_fn` to include `torch` in `fn_modules`

Reply:
Updated, thanks!